src/middlewared/middlewared/plugins/init_shutdown_script.py

```diff
@@ -132,27 +145,50 @@ class Config:
     async def execute_task(self, task):
         task_type = task['type']
         cmd = None
+        tmp_script = None
 
         if task_type == 'COMMAND':
             cmd = task['command']
+        elif task_type == 'SCRIPT' and task['script_text']:
+            _, tmp_script = tempfile.mkstemp(text=True)
+            os.chmod(tmp_script, stat.S_IRWXU)
+            with open(tmp_script, 'w') as f:
+                f.write(task['script_text'])
+            cmd = f'exec {tmp_script}'
         elif os.path.exists(task['script'] or '') and os.access(task['script'], os.X_OK):
             cmd = f'exec {task["script"]}'
 
-        if cmd:
-            proc = await Popen(
-                cmd,
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.PIPE,
-                shell=True,
-                close_fds=True
-            )
-            stdout, stderr = await proc.communicate()
-
-            if proc.returncode:
-                self.middleware.logger.debug(
-                    'Execution failed for '
-                    f'{task_type} {task["command"] if task_type == "COMMAND" else task["script"]}: {stderr.decode()}'
+        try:
+            if cmd:
+                proc = await Popen(
+                    cmd,
+                    stdout=subprocess.DEVNULL,
+                    stderr=subprocess.PIPE,
+                    shell=True,
+                    close_fds=True
                 )
+                stdout, stderr = await proc.communicate()
+
+                if proc.returncode:
+                    if task_type == 'COMMAND':
+                        cmd = task['command']
+                    elif task_type == 'SCRIPT' and task['script_text']:
+                        cmd = task['script_text']
+                    elif task_type == 'SCRIPT' and task['script']:
+                        cmd = task['script']
+                    else:
+                        cmd = ''
+                    self.middleware.logger.debug(
+                        'Execution failed for '
+                        f'{task_type} {cmd}: {stderr.decode()}'
```

Comment:
`script_text` can be big multi-line string, we don't want that in logs.
Why only stderr? Won't it be better to redirect stderr to stdout and put stdout here?

Reply:
It will still be multi-line and confusing. As we don't have any other visible identifier, I would go with `ellipsis(repr(task['script_text'], 64))`

Reply:
> It will still be multi-line and confusing. As we don't have any other visible identifier, I would go with `ellipsis(repr(task['script_text'], 64))`

Not sure what you mean.
`script_text` will be always multi-line.
I guess putting the first `script_text` lines to the log will not help a lot to identify the problematic `script_text`.
So I'm not sure what to do.

Reply:
Put some identifier of the task (e.g. id and comment). We dont need the script text.

Reply:
Let's assume that first 64 chars of script (in one line) is the best we can provide to identify it, everything else will be flooding logs.

Reply:
@william-gr ID is useless (not seen in UI) and we don't have comment (maybe we should add one? `name`)

Reply:
Thats too verbose, we just need an identifier, the id and comment are good enough. The script it too verbose either way.

Reply:
Better than a multi line script. middlewared.log is not seen in the UI either so its a moot point.

Which raises the question, why are we even logging it instead of raising a CallError?

Reply:
I will asnwer that myself, to allow other scripts to run.

That needs to be refactored later on, but for the scope of this ticket the ID seems good enough to me. (We can eventualyl show the ID in the UI if thats the case)

Reply:
@william-gr because now it's the only way for these to be seen, especially for those at SHUTDOWN.

As we are in 12.0, maybe we can think if it is worth to store exit code / output of the last run in DB and expose it in UI?

Reply:
We can store them in tmpfs as we do with other job logs, but there still there will be no way to retrieve errors/output for scripts ran at SHUTDOWN.

Reply:
@themylogin we do have a comment in 12.

Everything else seems out of the scope of the ticket to me, which can be improved later.